source/renderer/app/features/discreet-mode/ui/DiscreetValue.story.js

```diff
@@ -1,30 +1,44 @@
 // @flow
-import React from 'react';
+import React, { useLayoutEffect } from 'react';
 import { storiesOf } from '@storybook/react';
+import { observer } from 'mobx-react';
 import { withKnobs, boolean } from '@storybook/addon-knobs';
+import StoryDecorator from '../../../../../../storybook/stories/_support/StoryDecorator';
+import StoryProvider from '../../../../../../storybook/stories/_support/StoryProvider';
 import {
   DiscreetModeFeatureProvider,
   useDiscreetModeFeature,
 } from '../context';
 import DiscreetValue from './DiscreetValue';
 
-function Toggle({ knob }: { knob: boolean }) {
+const Toggle = observer(({ knob }: { knob: boolean }) => {
   const feature = useDiscreetModeFeature();
 
-  if (feature.isDiscreetMode !== knob) {
-    feature.toggleDiscreetMode();
-  }
+  useLayoutEffect(() => {
+    if (knob !== feature.isDiscreetMode) {
+      feature.toggleDiscreetMode();
+    }
+  }, [knob, feature.isDiscreetMode]);
 
   return null;
-}
+});
 
-storiesOf('Discreet Mode|Discreet Value', module)
+storiesOf('Discreet Mode|Discreet Asset Amount', module)
   .addDecorator(withKnobs)
-  .add('Main', () => (
-    <DiscreetModeFeatureProvider>
-      <>
-        <DiscreetValue>123</DiscreetValue>
-        <Toggle knob={boolean('Toggle', false)} />
-      </>
-    </DiscreetModeFeatureProvider>
+  .addDecorator((story) => <StoryDecorator>{story()}</StoryDecorator>)
+  .addDecorator((story) => <StoryProvider>{story()}</StoryProvider>)
+  .addDecorator((story) => (
+    <DiscreetModeFeatureProvider>{story()}</DiscreetModeFeatureProvider>
+  ))
```

Comment:
I guess you could merge these three `addDecorator` calls into a single one & wrap the `story()` with the three provider components via the markup.